### ISSUE TITLE
[Bug?]: The lockfile would have been modified by this install, which is explicitly forbidden

### ISSUE THREAD

**Issue by natew** [yarnpkg/berry#3804]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

At this commit or one of the two before it an error occurred only in CI:

https://github.com/tamagui/tamagui/commit/f72a2645a07a1ef9ee6719e464e1c28798cb52be

```
Post-resolution validation
82
  ➤ YN0000: │ @@ -20311,13 +20311,12 @@
83
  ➤ YN0000: │    linkType: hard
84
  ➤ YN0000: │  
85
  ➤ YN0000: │  "typescript@patch:typescript@npm%3A4.6.0-dev.20211126#~builtin<compat/typescript>":
86
  ➤ YN0000: │    version: 4.6.0-dev.20211126
87
  ➤ YN0028: │ -  resolution: "typescript@patch:typescript@npm%3A4.6.0-dev.20211126#~builtin<compat/typescript>::version=4.6.0-dev.20211126&hash=ddd1e8"
88
  ➤ YN0028: │ +  resolution: "typescript@patch:typescript@npm%3A4.6.0-dev.20211126#~builtin<compat/typescript>::version=4.6.0-dev.20211126&hash=493e53"
89
  ➤ YN0000: │    bin:
90
  ➤ YN0000: │      tsc: bin/tsc
91
  ➤ YN0000: │      tsserver: bin/tsserver
92
  ➤ YN0028: │ -  checksum: d3ab3a5463226e7870594f0b9b48a1da6972f2123eb1c79cc4d57c3586f138a23a2b50ec963902c26bf11a751e475de6dc750c64d704cfdcd635465599adf27e
93
  ➤ YN0000: │    languageName: node
94
  ➤ YN0000: │    linkType: hard
95
  ➤ YN0000: │  
96
  ➤ YN0000: │  "ua-parser-js@npm:^0.7.18, ua-parser-js@npm:^0.7.30":
97
  ➤ YN0000: │ 
98
  ➤ YN0028: │ The lockfile would have been modified by this install, which is explicitly forbidden.
99
➤ YN0000: └ Completed
```

You can see the run here: https://github.com/tamagui/tamagui/runs/4337635376?check_suite_focus=true

You can fork the repo and try it on that commit, I've since added `YARN_ENABLE_IMMUTABLE_INSTALLS=false` for now, but seems like a Yarn bug. Locally it installs fine with no problem, only on Github Actions it seems to have a problem. Perhaps when I added `@tamagui/ow` which seems to be the causative commit.

### To reproduce

```
git clone git@github.com:tamagui/tamagui.git
git checkout f72a2645a07a1ef9ee6719e464e1c28798cb52be
# push to a new github repo with actions to see it fail in CI
```

### Environment

```shell
Github actions CI runner
```

**Comment by natew** [comment]
If you check the repo you can see it fails with any version of TS, seems the version isn't the issue.

**Comment by merceyz** [comment]
The problem is that locally you're using Corepack which you've locked to use [`yarn@3.1.0`](https://github.com/tamagui/tamagui/blob/f4bf3f0749094cbdd420f99f9c0065cfeea2743d/package.json#L64) while on your CI [you always grab the latest version](https://github.com/tamagui/tamagui/blob/f4bf3f0749094cbdd420f99f9c0065cfeea2743d/.github/workflows/release.yaml#L30). We released [3.1.1 earlier today](https://github.com/yarnpkg/berry/releases/tag/%40yarnpkg%2Fcli%2F3.1.1) which means your CI automatically upgraded.
